File: 283-move-zeroes/move.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        nonZeroes=0
        for i in range(len(nums)):
            if nums[i] !=0:
                nums[nonZeroes], nums[i] = nums[i], nums[nonZeroes] # swap positions if nums[i] != 0 
                nonZeroes +=1

# first solution: less optimized
    # def moveZeroes(self, nums: List[int]) -> None:
    #     """
    #     Do not return anything, modify nums in-place instead.
    #     """
    #     nonZeroes = 0 
    #     for i in range(len(nums)):
    #         if nums[i]!=0:
    #             nums[nonZeroes] = nums[i]
    #             nonZeroes+=1

    #     for i in range(nonZeroes,len(nums)):
    #         nums[i] = 0
        

File: 283-move-zeroes/move.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */

 var moveZeroes = function(nums) {
    let nonZeroes = 0 
    for (let i = 0; i < nums.length; i++){
        let val = nums[i]
        nums[i] = 0 
        if (val!=0){
            nums[nonZeroes] = val
            nonZeroes++
        }
    }  
};

// original solution:
//  var moveZeroes = function(nums) {
//     let nonZeroes = 0 
//     for (let i = 0; i < nums.length; i++){
//         if (nums[i]!=0){
//             nums[nonZeroes] = nums[i]
//             nonZeroes++
//         }
//     }
    
//     for (let i =nonZeroes;i < nums.length;i++ ){
//         nums[i] = 0
//     }
// };